src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -498,77 +499,84 @@ FIXTURE_TEST(test_time_based_eviction, storage_test_fixture) {
     ss::abort_source as;
     storage::log_manager mgr = make_log_manager(cfg);
     info("Configuration: {}", mgr.config());
+    BOOST_REQUIRE(feature_table.local().is_active(
+      features::feature::broker_time_based_retention));
     auto deferred = ss::defer([&mgr]() mutable { mgr.stop().get0(); });
     auto ntp = model::ntp("default", "test", 0);
 
     storage::ntp_config ntp_cfg(ntp, mgr.config().base_dir);
-    auto log = mgr.manage(std::move(ntp_cfg)).get0();
-    auto disk_log = log;
+    auto disk_log = mgr.manage(std::move(ntp_cfg)).get0();
+
+    // keep track of broker timestamp, to predict expected evictions
+    auto broker_t0 = model::timestamp_clock::now();
+    constexpr static auto broker_ts_sep = 5s;
 
     // 1. segment timestamps from 100 to 110
+    // broker timestamp: broker_t0
     append_custom_timestamp_batches(
-      log, 10, model::term_id(0), model::timestamp(100));
+      disk_log, 10, model::term_id(0), model::timestamp(100));
     disk_log->force_roll(ss::default_priority_class()).get0();
 
     // 2. segment timestamps from 200 to 230
+    // b ts: broker_t0 + broker_ts_sep
+    ss::sleep(broker_ts_sep).get();
     append_custom_timestamp_batches(
-      log, 30, model::term_id(0), model::timestamp(200));
-
+      disk_log, 30, model::term_id(0), model::timestamp(200));
     disk_log->force_roll(ss::default_priority_class()).get0();
+
     // 3. segment timestamps from 231 to 250
+    // b_ts: broker_t0 + broker_ts_sep + broker_ts_sep
+    ss::sleep(broker_ts_sep).get();
     append_custom_timestamp_batches(
-      log, 20, model::term_id(0), model::timestamp(231));
+      disk_log, 20, model::term_id(0), model::timestamp(231));
 
     /**
      *  Log contains 3 segments with following timestamps:
-     *
+     * t0         t0+sep    t0+2*sep
      * [100..110][200..230][231..261]
      */
 
     storage::housekeeping_config ccfg_no_compact(
-      model::timestamp(200),
+      model::to_timestamp(broker_t0 - 10s),
```

Comment:
nit: `broker_ts_sep * 2` instead of `10s`

Reply:
done